src/predicates.cpp

```diff
@@ -117,4 +148,64 @@ void init_predicates(py::module& m) {
             Geography object(s)
 
     )pbdoc");
+
+    m.def("touches",
+          py::vectorize(TouchesPredicate()),
+          py::arg("a"),
+          py::arg("b"),
+          R"pbdoc(
+        Returns True if A and B intersect, but their interiors do not intersect.
+        A and B have at least one point in common, where the common point
+        lies in at least one boundary.
+
+        Parameters
+        ----------
+        a, b : :py:class:`Geography` or array_like
+            Geography object(s)
+
+    )pbdoc");
+
+    S2BooleanOperation::Options closed_options;
+    closed_options.set_polyline_model(S2BooleanOperation::PolylineModel::CLOSED);
+    closed_options.set_polygon_model(S2BooleanOperation::PolygonModel::CLOSED);
+
+    m.def("covers",
+          py::vectorize(Predicate(
+              [](const s2geog::ShapeIndexGeography& a_index,
+                 const s2geog::ShapeIndexGeography& b_index,
+                 const S2BooleanOperation::Options& options) {
+                  return s2geog::s2_contains(a_index, b_index, options);
+              },
+              closed_options)),
+          py::arg("a"),
+          py::arg("b"),
+          R"pbdoc(
+        Returns True if every point in B lies inside the interior or boundary of A.
+
+        Parameters
+        ----------
+        a, b : :py:class:`Geography` or array_like
+            Geography object(s)
+
```

Comment:
```suggestion

        Notes
        -----
        If A and B are both polygons and share co-linear edges,
        `covers` currently returns expected results only when those
        shared edges are identical.
 
```

Just to better notify users about this as it is not clear if / how easy it will be to work around it.

(an `Examples` section may better illustrate that behavior but it is something that we can address later more generally across the library)